### PR TITLE
Dev env setup

### PR DESCRIPTION
**Pull request by hiromipaw**
Addressing issue https://github.com/mozilla-services/cliquet/issues/65. Add paragraph on Contributing documentation on how to prepare your environment for developing Cliquet.

### CODE REVIEW

**Comment by hiromipaw** [comment]
Trying to document possible issues. Please feel free to jump in and comment in the meantime if you have any advice or you think I should consider something that I have completely overlooked.

File: docs/contributing.rst [hunk 1]
```diff
@@ -2,6 +2,44 @@ Contributing
...
+
+(deactivate when finished)
+
+Install packages as usual, for example install needed packages:
+
+::
+
+    $ pip install cliquet
+    $ pip install waitress
+    $ pip install tox
```
Comment (ametaireau) [inline docs/contributing.rst]: you can actually install everything with the dev-requirement.txt file. `pip install -r dev-requirements.txt`

File: docs/contributing.rst [hunk 2]
```diff
@@ -2,6 +2,42 @@ Contributing
...
+
+To prepare your system with Postgres and Redis please refer to the Installation guide.
+
+Prepare your project virtualenv:
+
+::
+
+    $ pip install virtualenv
+    $ virtualenv .venv
+    $ source .venv/bin/activate
```
Comment (ametaireau) [inline docs/contributing.rst]: Normally this is done automatically when doing a "make install".

Comment (ametaireau) [inline docs/contributing.rst]: So I'm not sure this is so valuable to put here.

Comment (leplatrem) [inline docs/contributing.rst]: ``make install-dev`` I would say

File: docs/contributing.rst [hunk 3]
```diff
@@ -2,6 +2,42 @@ Contributing
...
+
+    $ git clone https://github.com/mozilla-services/kinto.git
+
+
+Prepare Kinto to run:
+
+::
+
+    $ cd kinto
+    $ make serve
```
Comment (leplatrem) [inline docs/contributing.rst]: I suggest we add ``kinto`` to *dev-requirements.txt*, and add ``make runkinto`` here

**Comment by leplatrem** [comment]
Thank you very much!

You're the first external contributor and I saw that you're located in Barcelona :) I thus would like to invite you for some lunch/drink/tapas :tada: !

**Comment by hiromipaw** [comment]
Hey guys. Thanks for feedback. I'll make sure to add that. Hope to start contributing more :)
@leplatrem drinks/lunch/tapas in Barcelona sounds good. Are you passing by or do you live here?

S

**Comment by leplatrem** [comment]
I just started on a fresh Ubuntu 14.10 box, here is my log : 

```
$ make runkinto
...
pkg_resources.DistributionNotFound: kinto
```


```
$ source .venv/bin/activate
$ pip install kinto
```

Now I can run it : 
```
$ make runkinto
/home/mathieu/Code/Mozilla/cliquet/.venv/bin/pserve cliquet/tests/config/kinto.ini
2015-03-23 21:16:58,121 INFO  [cliquet.storage.postgresql][MainThread] Created PostgreSQL storage tables 
Starting server in PID 27106.
serving on http://0.0.0.0:8888
```

Still, it keeps asking for kinto...

```
$ make tests
Run 'make runkinto' before starting tests.
Makefile:37: recipe for target 'need-kinto-running' failed
```

Because, curl is not installed by default : 

``$ sudo apt-get install curl``

```
$ make tests
tox
make: tox: Command not found
Makefile:34: recipe for target 'tests' failed
make: *** [tests] Error 127
```

Install tox then!

```
$ pip install tox
Downloading/unpacking tox
  Downloading tox-1.9.1.tar.gz (93kB): 93kB downloaded
  Running setup.py (path:/tmp/pip-build-GI7i1u/tox/setup.py) egg_info for package tox
    
Downloading/unpacking virtualenv>=1.11.2 (from tox)
  Downloading virtualenv-12.0.7-py2.py3-none-any.whl (1.8MB): 1.8MB downloaded
Downloading/unpacking py>=1.4.17 (from tox)
  Downloading py-1.4.26.tar.gz (190kB): 190kB downloaded
  Running setup.py (path:/tmp/pip-build-GI7i1u/py/setup.py) egg_info for package py
    
Installing collected packages: tox, virtualenv, py
  Running setup.py install for tox
    
    Installing tox script to /home/mathieu/Code/Mozilla/cliquet/.venv/bin
    Installing tox-quickstart script to /home/mathieu/Code/Mozilla/cliquet/.venv/bin
  Running setup.py install for py
    
Successfully installed tox virtualenv py
Cleaning up...
```

Run tests ? Hey, young padawan, not so fast :)

```
$ make tests
tox
Traceback (most recent call last):
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/bin/tox", line 9, in <module>
    load_entry_point('tox==1.9.1', 'console_scripts', 'tox')()
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 521, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2632, in load_entry_point
    return ep.load()
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2312, in load
    return self.resolve()
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2318, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/tox/__init__.py", line 23, in <module>
    from tox._cmdline import main as cmdline  # noqa
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/tox/_cmdline.py", line 15, in <module>
    from tox._venv import VirtualEnv
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/tox/_venv.py", line 6, in <module>
    from tox._config import DepConfig
  File "/home/mathieu/Code/Mozilla/cliquet/.venv/local/lib/python2.7/site-packages/tox/_config.py", line 12, in <module>
    from pip.req.req_file import parse_requirements
ImportError: No module named req_file
Makefile:34: recipe for target 'tests' failed
make: *** [tests] Error 1
```

Upgrade pip : 

```
$ pip install -U pip
Downloading/unpacking pip from https://pypi.python.org/packages/py2.py3/p/pip/pip-6.0.8-py2.py3-none-any.whl#md5=41e73fae2c86ba2270ff51c1d86f7e09
  Downloading pip-6.0.8-py2.py3-none-any.whl (1.3MB): 1.3MB downloaded
Installing collected packages: pip
  Found existing installation: pip 1.5.6
    Uninstalling pip:
      Successfully uninstalled pip
Successfully installed pip
Cleaning up...
```

Yes. Now, you can.

```
$ make tests
tox
GLOB sdist-make: /home/mathieu/Code/Mozilla/cliquet/setup.py
py27 create: /home/mathieu/Code/Mozilla/cliquet/.tox/py27
py27 installdeps: coverage, mock, nose, psycopg2, raven, statsd, unittest2, webtest
py27 inst: /home/mathieu/Code/Mozilla/cliquet/.tox/dist/cliquet-1.3.3.dev0.zip
py27 runtests: PYTHONHASHSEED='2218185105'
py27 runtests: commands[0] | python --version
Python 2.7.9
....
```

``fatal error: Python.h: No such file or directory`` with Python3.4 ? oh, forgot this one :)

``sudo apt-get install python3.4-dev``

**Comment by hiromipaw** [comment]
Oh great! I wanted to try this out just today. On OSX is a bit more complicated and I have been delayed. Will update tomorrow if that's ok.

**Comment by hiromipaw** [comment]
Oh great! I wanted to try this out just today. On OSX is a bit more complicated and I have been delayed. Will update tomorrow if that's ok.

**Comment by hiromipaw** [comment]
Oh great! I wanted to try this out just today. On OSX is a bit more complicated and I have been delayed. Will update tomorrow if that's ok.

**Comment by ametaireau** [comment]
sure, just let us know :)

**Comment by hiromipaw** [comment]
Sorry for being late. Got sick this week and got delayed :) I have updated the docs. On a side note I would also add tox to the dev-requirements.txt

**Comment by ametaireau** [comment]
Thanks a lot! It's now merged :-)

**Pull request merged**